hkmc2/shared/src/main/scala/hkmc2/codegen/Deforestation.scala

```
@@ -415,19 +428,19 @@ class Deforest(using TL, Raise, Elaborator.State):
         case FunDefn(_, sym, params, body) =>
           val funSymStratVar = symToStrat(sym)
           val param = params.head match
-            case ParamList(flags, params, restParam) => params
-          val funStrat = constrFun(param, body) // TODO: handle mutiple param list
+            case ParamList(flags, params, N) => params // TODO: handle mutiple param list
```

You're still silently discarding extra parameter lists!!

---

Sorry I missed this earlier, now it only works when `params` has exactly one element, and that element's `restParam` is `None`. Otherwise it crashes instead of silently doing the wrong thing.